From marcin.slusarz@gmail.com Sat Dec 15 18:21:46 2007
From: Marcin Slusarz <marcin.slusarz@gmail.com>
Date: Sun, 16 Dec 2007 03:23:20 +0100
Subject: USB: usbdevfs_urb: __user annotation
To: linux-kernel@vger.kernel.org
Cc: Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <20071216022317.GI26986@joi>
Content-Disposition: inline


fix warning:
drivers/usb/core/devio.c:1226:20: warning: incorrect type in assignment (different address spaces)
drivers/usb/core/devio.c:1226:20:    expected void *usercontext
drivers/usb/core/devio.c:1226:20:    got void [noderef] <asn:1>*

Signed-off-by: Marcin Slusarz <marcin.slusarz@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/usbdevice_fs.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/usbdevice_fs.h
+++ b/include/linux/usbdevice_fs.h
@@ -104,7 +104,7 @@ struct usbdevfs_urb {
 	int error_count;
 	unsigned int signr;	/* signal to be sent on completion,
 				  or 0 if none should be sent. */
-	void *usercontext;
+	void __user *usercontext;
 	struct usbdevfs_iso_packet_desc iso_frame_desc[0];
 };
 
